.pre-commit-config.yaml

```diff
@@ -51,12 +51,17 @@ repos:
     files: \.py$
 
 - repo: https://github.com/pre-commit/mirrors-mypy
-  rev: v0.800
+  rev: v0.910
   hooks:
   - id: mypy
     exclude: tests
-    args: ["--pretty"]
-    additional_dependencies: ['nox==2020.12.31']
+    args: ["--pretty", "--show-error-codes"]
+    additional_dependencies: [
+        'keyring==23.0.1',
+        'nox==2020.12.31',
+        'types-docutils==0.1.8',
+        'types-six==0.1.9',
```

Comment:
Do we use `six` directly anywhere? I don’t think this covers `pip._vendor.six`, would it?

Reply:
Was seeing this
```
src/pip/_internal/operations/install/wheel.py:41: error: Module
"pip._vendor.six" has no attribute "ensure_str"  [attr-defined]
    from pip._vendor.six import ensure_str, ensure_text, reraise
```

I believe this happens because we have a mypy stub file to redirect our vendored code to the six package stubs https://github.com/pypa/pip/blob/main/src/pip/_vendor/six/__init__.pyi

With the recent advent of modular typeshed, we have to explicitly include `types-six` dependency as the typeshed types for six ship separately from mypy itself. (This is a benefit IMO).

Reply:
If it helps, six is being removed from this file in #10260 so maybe once that lands, including `types-six` will not be necessary.

Reply:
> mypy stub file to redirect our vendored code to the six package stubs

Thanks, that makes sense. I don’t know if that’s the intention (I think so considering Pradyun is much more knowledgable on this than I am), but it’s a good design :)